### PR TITLE
Handling the large video to specify the size of chunk

### PR DESCRIPTION
**Pull request by motoyasu-yamada**

### CODE REVIEW

**Comment by JuanIrache** [comment]
Hello @motoyasu-yamada Thanks for the PR, this seems to be great progress. However, there's the problem that the module was initially conceived to work in the browser. Using `const fs = require('fs');` on **index.js** breaks compatibility with the browser. Could you suggest a way to overcome this? One way could be to put all the logic inside `if (typeof file === "string") {` in a separate function that only NodeJS uses. What do you think?

**Comment by motoyasu-yamada** [comment]
Thanks. But I’m sorry for my mistake. I will do that tomorrow.

**Comment by motoyasu-yamada** [comment]
Removed codes using "fs" module in "index.js". Please check index.test.js and README.md

File: index.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ module.exports = function (file, isBrowser = false, update, options) {
   var workerRunning = true;
   return new Promise(function (resolve, reject) {
     var readBlock = readBlockFactory();
-    mp4boxFile = MP4Box.createFile(false);
+    mp4boxFile = MP4Box.createFile();
```
Comment (JuanIrache) [inline index.js:35]: Why is the **false** removed here? I don't remember why it was there in the fist place, but just to be sure that we are not breaking anything. Thank you

**Comment by motoyasu-yamada** [comment]
If false is specified in the first argument of `createFile`, `onSamples` will not fire when parsing a huge video file.

The documentation officially has no arguments.
https://gpac.github.io/mp4box.js/

Looking at the source code, if the first argument is false, `discardMdatData` will be true
https://github.com/gpac/mp4box.js/blob/master/src/mp4box.js

`discardMdatData` is mainly used below.
https://github.com/gpac/mp4box.js/blob/47c77fe14c7f8f16c380b2092a252c080b17a80c/src/isofile-advanced-parsing.js

```js
ISOFile.prototype.updateUsedBytes = function(box, ret) {
if (this.stream.addUsedBytes) {
if (box.type === "mdat") {
/* for an mdat box, only its header is considered used, other bytes will be used when sample data is requested */
this.stream.addUsedBytes(box.hdr_size);
if (this.discardMdatData) {
this.stream.addUsedBytes(box.size-box.hdr_size);
}
```

On the contrary, why specify false?

**Comment by JuanIrache** [comment]
> If false is specified in the first argument of `createFile`, `onSamples` will not fire when parsing a huge video file.
> 
> The documentation officially has no arguments.
> https://gpac.github.io/mp4box.js/
> 
> Looking at the source code, if the first argument is false, `discardMdatData` will be true
> https://github.com/gpac/mp4box.js/blob/master/src/mp4box.js
> 
> `discardMdatData` is mainly used below.
> https://github.com/gpac/mp4box.js/blob/47c77fe14c7f8f16c380b2092a252c080b17a80c/src/isofile-advanced-parsing.js
> 
> ```js
> ISOFile.prototype.updateUsedBytes = function(box, ret) {
> if (this.stream.addUsedBytes) {
> if (box.type === "mdat") {
> /* for an mdat box, only its header is considered used, other bytes will be used when sample data is requested */
> this.stream.addUsedBytes(box.hdr_size);
> if (this.discardMdatData) {
> this.stream.addUsedBytes(box.size-box.hdr_size);
> }
> ```
> 
> On the contrary, why specify false?

I seem to remember I had problems when false was not specified, back when I created this, but I can't reproduce the issue. I'll let it through and be vigilant for problems. Thanks for the progress you've made!

**Comment by Motoyasu-Yamada-Spicysoft** [comment]
New patch is released. Fully fixed. 

https://github.com/gpac/mp4box.js/issues/205
->
https://github.com/gpac/mp4box.js/commit/2bfb7a53d8a66db846c26375c2693de1b7cd4c5b

**Pull request merged**